### PR TITLE
bugfix/20164-custom-dts

### LINKED ISSUE #20164: Declaration file (like highcharts.d.ts) missing for custom packages
#### Expected behaviour
After building a custom package as described here: [Documentation](https://www.highcharts.com/docs/getting-started/how-to-create-custom-highcharts-packages) I get type declarations just like using the regular package.

#### Actual behaviour
Instead, I get an error 
```
TS7016: Could not find a declaration file for module  @company/custom-highcharts .
/Users/.../packages/highcharts/code/es-modules/masters/custom.src.js
implicitly has an  any  type.
```

#### Product version

Highcharts 11.2.0

### PR DESCRIPTION
**Pull request by bre1470**
Fixed #20164; Doc for custom bundle contains solution for custom definition file.

### CODE REVIEW

File: tools/gulptasks/jsdoc-dts.js [hunk 1]
```diff
@@ -86,13 +87,19 @@ function jsDocESMDTS() {
...
     const highchartsDeclarationsGenerator = require(
         '@highcharts/highcharts-declarations-generator'
     );
 
     return new Promise((resolve, reject) => {
 
+        if (argv.custom) {
+            highchartsDeclarationsGenerator.config
+                .mainModule = 'code/custom';
+        }
```
Comment (goransle) [inline tools/gulptasks/jsdoc-dts.js:101]: I'm getting a `TypeError: Cannot set properties of undefined (setting 'mainModule
')` (even after removing node_modules and reinstalling)

**Comment by goransle** [comment]
Issue number in commit message should be #20164

**Review by goransle — APPROVED** [review]
Looks good

**Review by marvin19 — APPROVED** [review]

**Review by MarkusBarstad — COMMENTED** [review]
I am getting 
Screenshot 2024-01-17 at 09 49 59

 when running `npx gulp jsdoc-dts --custom` :grimacing:

I could be misunderstanding something about this custom declaration file feature; Do I need to supply a `ts/masters/custom.src.ts`, using HC stuff, and then I will be able to generate a custom decleration file? 

I have ran `npx gulp scripts --force` and `npx gulp scripts-compile`, and I have removed `node_modules` and am on a fresh `npm install`

**Comment by bre1470** [comment]
@MarkusBarstad 
> I could be misunderstanding something about this custom declaration file feature; Do I need to supply a `ts/masters/custom.src.ts`, using HC stuff, and then I will be able to generate a custom decleration file?

Yes exactly. This is an addition to the guide in the document. You have to follow the steps to reach a custom declaration. If you think this can be written in a better way, please let me know.

**Comment by MarkusBarstad** [comment]
> Yes exactly. This is an addition to the guide in the document. You have to follow the steps to reach a custom declaration. If you think this can be written in a better way, please let me know.

No problem, i think its well written!

**Review by MarkusBarstad — COMMENTED** [review]
🚀

**Review by MarkusBarstad — APPROVED** [review]
🚀

**Review by hubertkozik — APPROVED** [review]
I had to reinstall node_modules, but after that everything is working correctly!

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**